src/components/ActionDetail/ActionDetail.tsx

```diff
@@ -99,75 +102,79 @@ export function ActionDetail({
     setText(action.text);
   }
 
   return (
-    <EuiPanel style={{ margin: "16px 0px" }} hasShadow={false} paddingSize="s">
+    <EuiPanel hasBorder={false} hasShadow={false} paddingSize="none">
+      <EuiText>
+        <h4>Edit action</h4>
+      </EuiText>
+      <EuiSpacer />
       <EuiFlexGroup alignItems="baseline">
-        <EuiFlexItem grow={false} style={{ width: 50 }}>
-          {title !== "Assert" && <EuiText size="s">{title}</EuiText>}
-        </EuiFlexItem>
         {url && (
           <EuiFlexItem>
-            <EuiFieldText
+            <FormControl
+              name={action.name}
+              onChange={e => setUrl(e.target.value)}
               value={url}
-              onChange={e => onURLChange(e.target.value)}
-            ></EuiFieldText>
+            />
           </EuiFlexItem>
         )}
         {selector && !action.isAssert && (
           <EuiFlexItem>
-            <EuiFieldText
+            <FormControl
+              name={action.name}
+              onChange={e => setSelector(e.target.value)}
               value={selector}
-              onChange={e => onSelectorChange(e.target.value)}
-            ></EuiFieldText>
+            />
           </EuiFlexItem>
         )}
         {text && (
           <EuiFlexItem>
-            <EuiFieldText
+            <FormControl
+              label="Value"
+              name={action.name}
+              noPrepend
+              onChange={e => setText(e.target.value)}
               value={text}
-              onChange={e => onTextChange(e.target.value)}
-            ></EuiFieldText>
-          </EuiFlexItem>
-        )}
-        {!action.isAssert && (
-          <EuiFlexItem grow={false}>
-            <EuiToolTip content="Add an assertion">
-              <EuiButtonIcon
-                aria-label="Opens a dialogue to create an assertion after this action"
-                color="text"
-                iconType="plusInCircle"
-                onClick={() =>
-                  onShowAssertionDrawer({
-                    previousAction: actionContext,
-                    stepIndex,
-                    actionIndex,
-                    mode: "create",
-                  })
-                }
-              />
-            </EuiToolTip>
-          </EuiFlexItem>
-        )}
-        {action.isAssert && (
-          <EuiFlexItem>
-            <Assertion
-              action={action}
-              actionContext={actionContext}
-              actionIndex={actionIndex}
-              assertionCount={assertionCount}
-              onDeleteAction={onDeleteAction}
-              onShowAssertionDrawer={onShowAssertionDrawer}
-              stepIndex={stepIndex}
             />
           </EuiFlexItem>
         )}
       </EuiFlexGroup>
+      <EuiSpacer />
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            onClick={() => {
+              if (url) {
+                onURLChange(url);
```

Comment:
I think these names are chosen to avoid overloading the actual `Setter` functions instantiated above. Like we have `setText`, `setUrl`, etc., already defined in this file, then they're referenced by these `onXChange` handlers that have some extra logic nested in them before calling the raw `Setter` function.

I'm fine to change this if it makes it easier to read, but we'd have some naming collisions if we tried doing this without renaming the existing set functions.